File: src/main/java/random/telegramhomebot/bootstrap/BootstrapLoader.java
package random.telegramhomebot.bootstrap;

import lombok.RequiredArgsConstructor;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import random.telegramhomebot.services.MessageService;
import random.telegramhomebot.telegram.Bot;

import static random.telegramhomebot.AppConstants.Messages.CHATBOT_STARTED_MSG;

@RequiredArgsConstructor
@Component
public class BootstrapLoader implements CommandLineRunner {

	private final Bot bot;
	private final MessageService messageService;

	@Override
	public void run(String... args) {
		sendBootstrapMessage();
	}

	private void sendBootstrapMessage() {
		bot.sendMessage(messageService.getMessage(CHATBOT_STARTED_MSG));
	}
}

File: src/main/java/random/telegramhomebot/bootstrap/BootstrapLoader.kt
package random.telegramhomebot.bootstrap

import org.springframework.boot.CommandLineRunner
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import random.telegramhomebot.const.AppConstants.CHATBOT_STARTED_MSG
import random.telegramhomebot.services.messages.MessageService
import random.telegramhomebot.integrations.telegram.send.TelegramMessageSender
import random.telegramhomebot.utils.logger

@Order(99)
@Component
class BootstrapLoader(
    private val telegramMessageSender: TelegramMessageSender,
    private val messageService: MessageService
) : CommandLineRunner {
    val log = logger()

    override fun run(vararg args: String) {
        log.info("BootstrapLoader started")
        sendBootstrapMessage()
    }

    private fun sendBootstrapMessage() {
        telegramMessageSender.sendMessage(messageService.getMessage(CHATBOT_STARTED_MSG))
    }
}
